docs/Testing.md

```diff
@@ -116,35 +116,55 @@ Fastify can also be tested after starting the server with `fastify.listen()` or
 
 Uses **app.js** from the previous example.
 
-**test-listen.js** (testing with [`Request`](https://www.npmjs.com/package/request))
+Let's compare 2 examples using different test frameworks:
+
+**test-listen.js** (with [`Got`](https://www.npmjs.com/package/got) & [`Tap`](https://www.npmjs.com/package/tap))
 ```js
 const tap = require('tap')
-const request = require('request')
+const got = require('got')
 const buildFastify = require('./app')
 
-tap.test('GET `/` route', t => {
-  t.plan(5)
-  
+tap.test('GET `/` route', async t => {
   const fastify = buildFastify()
   
   t.tearDown(() => fastify.close())
   
-  fastify.listen(0, (err) => {
-    t.error(err)
-    
-    request({
-      method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port
-    }, (err, response, body) => {
-      t.error(err)
-      t.strictEqual(response.statusCode, 200)
-      t.strictEqual(response.headers['content-type'], 'application/json; charset=utf-8')
-      t.deepEqual(JSON.parse(body), { hello: 'world' })
-    })
-  })
+  await fastify.listen(0)
+  const response = await got(`http://localhost:${fastify.server.address().port}`)
+  t.strictEqual(response.statusCode, 200)
+  t.strictEqual(response.headers['content-type'], 'application/json; charset=utf-8')
+  t.deepEqual(JSON.parse(response.body), { hello: 'world' })
+  t.end()
 })
 ```
 
+**test-listen.js** (with [`Got`](https://www.npmjs.com/package/got) & [`Mocha`](https://www.npmjs.com/package/mocha))
+```js
+const got = require('got')
+const buildFastify = require('./app')
+const assert = require('assert').strict;
+
+const fastify = buildFastify();
```

Comment:
This is problematic. A new instance of fastify should be started for every test, otherwise there is a significant risk of flaky tests.